packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationBackgroundService.kt

```diff
@@ -0,0 +1,155 @@
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.collections.ArrayDeque
+
+
+class PushNotificationBackgroundService : MethodChannel.MethodCallHandler, JobIntentService() {
+
+    /**
+     * The work queue
+     */
+    private val queue = ArrayDeque<List<Any>>()
+
+    /**
+     * The Background Channel that is used to send messages to the callback dispatcher
+     */
+    private lateinit var backgroundChannel: MethodChannel
+
+    companion object {
+
+        private const val TAG = "PushBackgroundService"
+
+        @JvmStatic
+        private val JOB_ID = UUID.randomUUID().mostSignificantBits.toInt()
+
+        @JvmStatic
+        private var backgroundFlutterEngine: FlutterEngine? = null
+
+        @JvmStatic
+        private val serviceStarted = AtomicBoolean(false)
+
+        @JvmStatic
+        fun enqueueWork(context: Context, work: Intent) {
+            enqueueWork(context, PushNotificationBackgroundService::class.java, JOB_ID, work)
+        }
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        startPushNotificationService(this)
+    }
+
+    private fun startPushNotificationService(context: Context) {
+        synchronized(serviceStarted) {
+            if (backgroundFlutterEngine == null) {
+
+                val callbackHandle = context.getSharedPreferences(
+                    PushNotificationConstants.SHARED_PREFERENCES_KEY, Context.MODE_PRIVATE
+                ).getLong(
+                    PushNotificationConstants.CALLBACK_DISPATCHER_HANDLE_KEY, 0
+                )
+                if (callbackHandle == 0L) {
+                    Log.w(TAG, "Warning: no callback registered")
+                    return
+                }
+                val callbackInfo =
+                    FlutterCallbackInformation.lookupCallbackInformation(callbackHandle)
+                if (callbackInfo == null) {
+                    Log.e(TAG, "Error: failed to find callback")
+                    return
+                }
+                Log.i(TAG, "Starting PushNotificationBackgroundService...")
+
+                // Create a background Flutter Engine
+                backgroundFlutterEngine = FlutterEngine(context)
+
+                val args = DartExecutor.DartCallback(
+                    context.assets, FlutterMain.findAppBundlePath(), callbackInfo
+                )
+
+                // DartCallback must only be executed after the engine has been created
+                backgroundFlutterEngine!!.dartExecutor.executeDartCallback(args)
+            }
+        }
+        backgroundChannel = MethodChannel(
+            backgroundFlutterEngine!!.dartExecutor.binaryMessenger,
+            "plugins.flutter.io/amplify_push_notification_plugin_background"
+        )
+        backgroundChannel.setMethodCallHandler(this)
+    }
+
+    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+        when (call.method) {
+            "PushNotificationBackgroundService.initialized" -> {
```

Comment:
I think this string should be treated as a method name. So exactly what is this method doing? Does this dot notation do anything special? Maybe this should just be `startBackgroundService` or something?

Reply:
dot doesn't do anything special. I renamed it to `callbackDispatcherInitialized` since that is what it signifies. It's called by the [Callback Dispatcher](https://github.com/aws-amplify/amplify-flutter/blob/base-project-setup-bg/packages/notifications/push/amplify_push_notifications/lib/callback_dispatcher.dart#L50) after initializing the background channel.